cfme/tests/infrastructure/test_advanced_search_providers.py

```diff
@@ -36,47 +35,63 @@ def hosts_with_vm_count(hosts):
 
 
 @pytest.yield_fixture(scope="function")
-def close_search():
-    """We must do this otherwise it's not possible to navigate after test!"""
+def setup_cleanup_search():
+    logger.debug('DEBUG FIXTURE: Starting, nav+clear search')
+    navigate_to(InfraProvider, 'All')
+    search.ensure_no_filter_applied()
+
     yield
+
+    # cleanup after test
+    logger.debug('DEBUG FIXTURE: Cleaning up search')
+    search.ensure_no_filter_applied()
     search.ensure_advanced_search_closed()
 
 
-pytestmark = [pytest.mark.usefixtures("close_search"), pytest.mark.tier(3)]
+@pytest.yield_fixture(scope="function")
+def rails_delete_filter(request):
+    yield
+    from fixtures import ssh_client
+    filter_name = getattr(request.function, "filter_name", None)
+    logger.debug('rails_delete_filter: calling rails to delete filter: {}'.format(filter_name))
+    if filter_name:
+        ssh_client.run_rails_command(
+            '"MiqSearch.where(:description => {}).first.delete"'.format(repr(filter_name)))
+    else:
+        logger.warning('rails_delete_filter: failed to get filter_name')
+
+pytestmark = [pytest.mark.usefixtures("setup_cleanup_search"), pytest.mark.tier(3)]
```

Comment:
for preventing later errors, please put pytestmark to the top of the file right after imports